File: app/Models/TipoHabitacion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class TipoHabitacion extends Model
{
    use HasFactory, SoftDeletes;

    protected $table = 'tipo_habitacion';
    protected $primaryKey = 'id_tipo_habitacion';

    protected $fillable = [
        'nombre_tipo',
        'descripcion',
        'capacidad_maxima',
    ];

    protected $dates = ['deleted_at'];
}

File: src/java/java/models/TipoHabitacion.java
package models;

import models.enums.NombreTipoHabitacion;

public class TipoHabitacion extends ClaseBase {
    private int idTipoHabitacion;
    
    private NombreTipoHabitacion nombreTipo;
    
    private String descripcion;
    
    private int capacidadMaxima;

    public int getIdTipoHabitacion() {
        return idTipoHabitacion;
    }

    public void setIdTipoHabitacion(int idTipoHabitacion) {
        this.idTipoHabitacion = idTipoHabitacion;
    }

    public NombreTipoHabitacion getNombreTipo() {
        return nombreTipo;
    }

    public void setNombreTipo(NombreTipoHabitacion nombreTipo) {
        this.nombreTipo = nombreTipo;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public int getCapacidadMaxima() {
        return capacidadMaxima;
    }

    public void setCapacidadMaxima(int capacidadMaxima) {
        this.capacidadMaxima = capacidadMaxima;
    }
}